core/encoding_jbig2.go

```diff
@@ -262,25 +314,38 @@ func newJBIG2DecoderFromStream(streamObj *PdfObjectStream, decodeParams *PdfObje
 			}
 		}
 	}
+	// if no decode params provided end fast.
+	if decodeParams == nil {
+		return encoder, nil
+	}
+	if globals := decodeParams.Get("JBIG2Globals"); globals != nil {
+		var err error
 
-	if decodeParams != nil {
-		if globals := decodeParams.Get("JBIG2Globals"); globals != nil {
-			var err error
-
-			globalsStream, ok := globals.(*PdfObjectStream)
-			if !ok {
-				err = errors.Error(processName, "jbig2.Globals stream should be an Object Stream")
-				common.Log.Debug("ERROR: %s", err.Error())
-				return nil, err
-			}
-			encoder.Globals, err = jbig2.DecodeGlobals(globalsStream.Stream)
-			if err != nil {
-				err = errors.Wrap(err, processName, "corrupted jbig2 encoded data")
-				common.Log.Debug("ERROR: %s", err)
-				return nil, err
-			}
+		globalsStream, ok := globals.(*PdfObjectStream)
+		if !ok {
+			err = errors.Error(processName, "jbig2.Globals stream should be an Object Stream")
+			common.Log.Debug("ERROR: %v", err)
+			return nil, err
+		}
+		encoder.Globals, err = jbig2.DecodeGlobals(globalsStream.Stream)
+		if err != nil {
+			err = errors.Wrap(err, processName, "corrupted jbig2 encoded data")
+			common.Log.Debug("ERROR: %v", err)
+			return nil, err
 		}
 	}
+	if k, err := GetNumberAsInt64(decodeParams.Get("Height")); err == nil {
```

Comment:
I think you could use the `UpdateParams` method here: `encoder.UpdateParams(decodeParams)`.